ttnn/cpp/ttnn/operations/reduction/prod/prod.cpp

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "prod.hpp"
+#include "prod_nc_op.hpp"
+#include "prod_op_all.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/auto_format.hpp"
+#include "ttnn/cpp/ttnn/operations/creation.hpp"
+#include "ttnn/operations/data_movement/slice/slice.hpp"
+#include "ttnn/operations/data_movement/permute/permute.hpp"
+#include "tt_numpy/functions.hpp"
+
+
+namespace ttnn {
+
+namespace operations {
+
+namespace reduction {
+
+inline Tensor zeros(
+    const tt::tt_metal::Shape shape, DataType data_type, Layout layout, Device* device, const MemoryConfig& output_mem_config) {
+    return tt::numpy::zeros(shape, data_type, layout, device, output_mem_config);
+}
+
+// Autoformat support
+inline Tensor change_layout_to_tile(const Tensor& temp, const MemoryConfig& output_mem_config) {
```

Comment:
I think I saw a similar function. Maybe this
https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.hpp

Reply:
looks like we are missing ttnn registration, but should be easy to reuse